### PR TITLE
Bl 937 be i want to add a syllabus to my course

### PR DESCRIPTION
**Pull request by macrowell70**
## Description

I added a field to the courses table for instructors to enter a URL that links to a syllabus for the respective course. I added the field to the knex schema, added a google docs URL to the seed data, updated the yup validation to confirm the input is a URL, and updated the README.md to show the syllabus_link field in the courses schema.

Fixes # (issue)

## Loom Video

https://www.loom.com/share/56848e48b3e84008b9c33d0f6ca8f47a

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by lisamdespain — DISMISSED** [review]
This looks really good. Seems like you hit every point, from migration to seeds, to the courses schema and README. One random question: _should_ the syllabus be a required field? I don't recall talking about that - it sounded more like it was optional. But there could have been more discussion about it that I missed. Regardless, code looks great!

**Comment by macrowell70** [comment]
@lisamdespain I have been wondering the same thing since I submitted this PR actually. We have a stakeholder meeting tomorrow, and I think I will ask Brianne whether she thinks it should be a required field.

**Comment by Austin-T-Johnson** [comment]
Great job, the code looks very clean and well implemented within the schema, seeds, and migration files! Also, good job updating the Readme!

**Review by jzkime — APPROVED** [review]
Nice creation of schema and relevant comments, plus additions to migrations and seeds. Everything seems well done.

File: data/migrations/20211102125418_courses.js [hunk 1]
```diff
@@ -40,6 +40,7 @@ exports.up = (knex) => {
       table.integer('number_of_sessions').notNullable();
       table.string('difficulty').notNullable();
       table.string('session_type').notNullable();
+      table.string('syllabus_link');
```
Comment (MaxT6) [inline data/migrations/20211102125418_courses.js:43]: In your video I thought you mentioned that the syllabus link was not nullable, but here it looks like it is, but maybe I am missing understanding the code?

File: data/seeds/009_courses.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ exports.seed = function (knex) {
       number_of_sessions: 4,
       difficulty: 'Easy',
       session_type: 'Group',
+      syllabus_link: 'https://docs.google.com',
```
Comment (MaxT6) [inline data/seeds/009_courses.js:21]: Does this force syllabuses to be on Google docs? Is there a way to make the link more universal, or is it not necessary?

**Review by MaxT6 — APPROVED** [review]
Great job adding a syllabus link to the migration and seeds data. Definitely a must have for any course. I left some questions for my own understanding but the code looks good to me!

**Review by Jwatk13 — APPROVED** [review]
Nice job remembering to update the ReadMe! Code is clean and easy to read.

**Review by vasilii-garanin — APPROVED** [review]
Hello Mitchel,
Good job creating the field to add syllabus link in courses. It's great how in comments you realized maybe syllabus should allow null, since it not be ready when instructor is initially creating the course but can always upload later when ready. I also agree with another comment maybe more options for file type would work better. Maybe instructor wants to upload pdf instead of doc. Nice work overall.

**Review by AlexiusThomas — COMMENTED** [review]
Wonderfully written ReadMe and great clear concise code.

**Pull request merged**